### PR TITLE
Finished 5.2 - cup of join exercise

### PR DESCRIPTION
**Pull request by EitanAb8**

### CODE REVIEW

File: cup_of_join.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+def join(*lists, sep="-"):
```
Comment (talbarfm) [inline cup_of_join.py]: give a full name - separator

File: cup_of_join.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    # This function receives lists and sep sign
+    # returns a new combined list with the sep sign between
+    # each list added.
+
+    if not len(lists):
+        return None
+    final_list = [inner for outer in lists[:len(lists)-1] for inner in outer+[sep]] + lists[len(lists)-1]
```
Comment (talbarfm) [inline cup_of_join.py]: you can just return the list comprehension

File: cup_of_join.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+def join(*lists, sep="-"):
+    # This function receives lists and sep sign
+    # returns a new combined list with the sep sign between
+    # each list added.
+
+    if not len(lists):
+        return None
```
Comment (talbarfm) [inline cup_of_join.py]: https://stackoverflow.com/questions/1839289/why-should-functions-always-return-the-same-type

File: cup_of_join.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
+def join(*lists, sep="-"):
+    # This function receives lists and sep sign
```
Comment (talbarfm) [inline cup_of_join.py]: Capture

Comment (talbarfm) [reply cup_of_join.py]: this is how I want you to document functions in python

File: cup_of_join.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    # returns a new combined list with the sep sign between
+    # each list added.
+
```
Comment (talbarfm) [inline cup_of_join.py]: you dont need this newline

File: cup_of_join.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    if not len(lists):
+        return None
+    final_list = [inner for outer in lists[:len(lists)-1] for inner in outer+[sep]] + lists[len(lists)-1]
+    return final_list
+
+
+def main():
+
```
Comment (talbarfm) [inline cup_of_join.py]: remove this newline its redundant

File: cup_of_join.py (hunk 2 above)
Comment (talbarfm) [inline cup_of_join.py]: spaces around arithmetic operators - not including the ones in the []

Comment (talbarfm) [inline cup_of_join.py]: lists[:len(lists)-1] is equivalent to lists[:-1]

Comment (talbarfm) [inline cup_of_join.py]: Nice nested list comprehension

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Comment by talbarfm** [comment]
Add type hints and return types to all of your functions as well

**Review by lovodkin93 — APPROVED** [review]
good job!

**Pull request merged**